File: MojioSDKiOS/MojioSDKiOS/Models/Location.m
//
//  Location.m
//  SdkTryout
//
//  Created by Ashish Agarwal on 2015-01-23.
//  Copyright (c) 2015 Ashish Agarwal. All rights reserved.
//

#import "Location.h"
#include <math.h>

@implementation Location

- (id) initWithLat:(double)latitude andLongitude:(double)longitude {
    self = [super init];
    if (self) {
        self.Lat = latitude;
        self.Lng = longitude;
    }
    
    return self;
}

-(BOOL) isValid {
    return !(isnan(self.Lat) || isnan(self.Lng)) && !(self.Lat == 0 && self.Lng ==0);
}

@end

File: sdkv2.0.0/sdkv2.0.0/MojioSDK/Location.swift
//
//  Location.swift
//  Motion
//
//  Created by Ashish Agarwal on 2016-02-11.
//  Copyright © 2016 Mojio. All rights reserved.
//

import UIKit
import ObjectMapper
import RealmSwift

class Location: Object, Mappable {
    dynamic var LocationAddress : Address? = nil
    dynamic var Timestamp : NSString? = nil
    dynamic var Lat : Float = 0
    dynamic var Lng : Float = 0
    dynamic var Status : NSString? = nil // ['Unknown', 'Locked', 'NotLocked', 'Predicted', 'DiffCorrected', 'LastKnown', 'TwoDFix', 'Historic', 'InvalidTime', 'CommunicationsFailure', 'GPSOff', 'PreviousValidState'],
    dynamic var Dilution : Float = 0
    dynamic var Altitude : Float = 0
    dynamic var GeoHash : NSString? = nil
    
    required convenience init?(_ map: Map) {
        self.init()
    }
    
    func mapping(map: Map) {
        LocationAddress <- map["Address"];
        Timestamp <- map["Timestamp"];
        Lat <- map["Lat"];
        Lng <- map["Lng"];
        Status <- map["Status"];
        Dilution <- map["Dilution"];
        Altitude <- map["Altitude"];
        GeoHash <- map["GeoHash"];
    }

}
